File: framework/languages/java/e2e/DiscoveryRegistryHa/src/main/java/systems/zlink/e2e/discoveryregistryha/handlers/WorkRequestHandler.java
package systems.zlink.e2e.discoveryregistryha.handlers;

import systems.zlink.e2e.discoveryregistryha.Contracts;
import systems.zlink.e2e.discoveryregistryha.ProviderState;
import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;

@ZLinkHandlerGroup(Contracts.HANDLER_GROUP)
public final class WorkRequestHandler
    implements ZLinkRequestHandler<Contracts.WorkRequest, Contracts.WorkReply> {
    private final ProviderState state;

    public WorkRequestHandler(ProviderState state) {
        this.state = state;
    }

    @Override
    public Contracts.WorkReply handle(
        Contracts.WorkRequest request,
        ZLinkRequestContext context) {
        return new Contracts.WorkReply("work:" + request.value(), state.providerRid());
    }
}

File: framework/languages/java/e2e-kotlin/DiscoveryRegistryHa/Server/Provider/src/main/kotlin/systems/zlink/e2e/kotlin/discoveryregistryha/provider/Handlers/WorkRequestHandler.kt
package systems.zlink.e2e.kotlin.discoveryregistryha.provider.Handlers

import systems.zlink.e2e.kotlin.discoveryregistryha.Contracts
import systems.zlink.e2e.kotlin.discoveryregistryha.provider.Support.ProviderEvidenceStore
import systems.zlink.framework.channels.ZLinkRequestContext
import systems.zlink.framework.channels.ZLinkRequestHandler
import systems.zlink.framework.handlers.ZLinkHandlerGroup

@ZLinkHandlerGroup(Contracts.HANDLER_GROUP)
class WorkRequestHandler(
    private val state: ProviderEvidenceStore,
) : ZLinkRequestHandler<Contracts.WorkRequest, Contracts.WorkReply> {
    override fun handle(
        request: Contracts.WorkRequest,
        context: ZLinkRequestContext,
    ): Contracts.WorkReply =
        Contracts.WorkReply("work:${request.value}", state.providerRid)
}
